### PR TITLE
Append name to rename target if it is a directory

### LINKED ISSUE #9266: Giving directory as a rename target uses directory as buffer-name
#### Description :octocat:
Giving directory as a rename target uses directory as buffer-name 

#### Reproduction guide :beetle:
- Start Emacs
- Visit a file say `/tmp/a.txt`
- `SPC f R` and give directory name as target say `/tmp2`

*Observed behaviour:* :eyes: :broken_heart:
- file `a.txt` is moved to `/tmp2/a.txt`
- buffer name is '/tmp2/' and it visits `/tmp2/a.txt`
- similar behaviour is observed in `spacemacs/rename-file` where buffer visits target directory

*Expected behaviour:* :heart: :smile:
- file `a.txt` is moved to `/tmp2/a.txt`
- buffer name is 'a.txt' and it visits `/tmp2/a.txt`

#### System Info :computer:
- OS: gnu/linux
- Emacs: 24.5.1
- Spacemacs: 0.200.9
- Spacemacs branch: master (rev. 8e1af145)
- Graphic display: t
- Distribution: spacemacs
- Editing style: hybrid
- Completion: helm
- Layers:
```elisp
(ansible graphviz auto-completion c-c   elfeed csv emacs-lisp erc
         git gtags haskell helm html java javascript latex markdown org php python
         (ruby :variables ruby-version-manager 'rvm)
         ruby-on-rails rust shell shell-scripts spell-checking sql syntax-checking typescript version-control xkcd yaml)
```

#### Backtrace :paw_prints:
```

```

### PR DESCRIPTION
**Pull request by shanavas786**
Fixes #9266

### CODE REVIEW

**Comment by Miciah** [comment]
Counsel upstream recently added a "move or rename" action that uses `dired-rename-file`: https://github.com/abo-abo/swiper/commit/944c20d41a09532757aa144c5cd124d33f0396c4.  Do we still need `spacemacs/rename-file`?

**Comment by d12frosted** [comment]
@Miciah Indeed. Didn't know about that function.

Though `spacemacs/rename-file` does much more. It also invalidates the projectile cache, renames associated buffer, cleanups recentf.

I am all for making the current solution to work with directories.

File: layers/+distributions/spacemacs-base/funcs.el [hunk 1]
```diff
@@ -332,6 +336,9 @@ initialized with the current filename."
                               (yes-or-no-p
                                (format "Create directory '%s'?" dir)))
                      (make-directory dir t)))
+                 ;; if new-name is a directory, append name to it
+                 (if (f-dir-p new-name)
```
[the remark this answers is not in the document]
Comment (d12frosted) [reply layers/+distributions/spacemacs-base/funcs.el]: Actually, I don't understand why do we need this here, in a function for renaming a buffer and associated file. It has nothing to do with directories, right?

[the remark this answers is not in the document]
Comment (shanavas786) [reply layers/+distributions/spacemacs-base/funcs.el]: What if the destination is a directory ?. this handles that condition

[the remark this answers is not in the document]
Comment (d12frosted) [reply layers/+distributions/spacemacs-base/funcs.el]: How can it be? I just don't get it. When you rename a buffer you select a name for it, and then the file visited by the buffer is being renamed. You don't deal with directories here. Am I missing something?

**Pull request closed without merging**